oll_sc/cli.py

```diff
@@ -34,30 +56,47 @@ def check_pin(pin):
 
 @oll_sc.command()
 @click.argument('input_data')
+@click.argument('key_id', type=int)
 @click.argument('pin')
-@click.option('--key_id')
 @click.option('--output-path', '-o', type=click.Path(), default=None,
               help='The output file path to write signature to.')
-def sign_rsa_pkcs_pss_sha256(input_data, pin, key_id=None, output_path=None):
+def sign_rsa_pkcs_pss_sha256(input_data, key_id, pin, output_path=None):
   """Sign input using SHA256_RSA_PKCS_PSS mechanism."""
-  if key_id is None:
-    key_id = (0x01,)
-
   # Read file if input_data is path
   try:
     input_data = Path(input_data).read_bytes()
   except IOError:
     pass
 
   try:
-    signature = sc_sign_rsa_pkcs_pss_sha256(input_data, pin, key_id)
+    signature = sc_sign_rsa_pkcs_pss_sha256(input_data, (key_id,), pin)
 
     if output_path:
       with open(output_path, 'wb') as out:
-        click.echo(type(signature))
         out.write(signature)
     else:
       click.echo(signature)
 
   except SmartCardError as e:
     click.echo(e)
+
+
+@oll_sc.command()
+@click.argument('key_id', type=int)
+@click.argument('pin')
+@click.option('--output-path', '-o', type=click.Path(), default=None,
+              help='The output file path to write public key pem to.')
+def x509(key_id, pin, output_path=None):
+  """Extract x509 certificate from smart card in PEM format."""
+  try:
+    x509_cert_bytes = sc_export_x509_pem((key_id,), pin)
+    x509_cert = x509_cert_bytes.decode('utf-8')
```

Comment:
Same here, could there be a function in the api which returns a string?

Reply:
Similar as https://github.com/openlawlibrary/oll-sc/pull/2#discussion_r288327632